<launch>
  <arg name="circle_simulating" default="false"/>
  <arg name="gazebo" default="true"/>
  <group if="$(arg circle_simulating)">
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find quadrotor_trajectory)/config/trajectory_visualization.rviz" />
    <node pkg="quadrotor_trajectory" type="simulate_truck_position.py" name="simulate_robot_position" output="screen" />
  </group>
  <node pkg="quadrotor_trajectory" type="truck_trajectory_estimator_node" name="truck_trajectory_estimator_node" output="screen" >
    <param name="truck_odom_sub_topic_name" type="string" value="/simulating_truck_odom" if="$(arg circle_simulating)" />
    <param name="truck_odom_sub_topic_name" type="string" value="/truck/ground_truth/odom" unless="$(arg circle_simulating)" />
    <param name="uav_odom_sub_topic_name" type="string" value="/ground_truth/state" if="$(arg gazebo)" />
    <param name="polynomial_order"  value="6" />
    <param name="derivation_order"  value="2" />
    <param name="solver_mode"  value="0" />
    <param name="lambda_D"  value="0.01" />
    <param name="estimating_odom_number"  value="10" />
    <param name="trajectory_generate_freqency"  value="30" />
    <param name="visualization_predict_time"  value="6.0" />
    <param name="smooth_forward_time"  value="1.0" />
    <param name="uav_vel_upper_bound"  value="10.0" />
    <param name="uav_vel_lower_bound"  value="-10.0" />
    <param name="uav_acc_upper_bound"  value="4.0" />
    <param name="uav_acc_lower_bound"  value="-4.0" />

  </node>
</launch>
